src/components/Client.ts

```diff
@@ -5,24 +5,39 @@ import {
   LanguageClientOptions,
   ServerOptions,
   DidOpenTextDocumentNotification,
-  DidSaveTextDocumentNotification
+  DidSaveTextDocumentNotification,
+  TransportKind
 } from "vscode-languageclient";
+import { Config } from "./Config";
+import { FileSystem } from "../executables/File";
 
 export class Client {
   private languageClient: LanguageClient;
+  private context: ExtensionContext;
 
   // constructor
-  constructor(lspExe: string, logFilePath: string) {
-    let cmd = lspExe;
-    let debugArgs = ["--disable-folding", "-l", logFilePath];
+  constructor(config: Config, logFilePath: string, context: ExtensionContext) {
+    this.context = context;
+    let runArgs = ["--disable-folding", "--ipc"];
+    let debugArgs = [...runArgs, "-l", logFilePath];
+    let dir = "dist/out";
+
+    let defaultModuleOptions = {
+      transport: TransportKind.ipc,
+      options: {
+        cwd: this.context.asAbsolutePath(dir)
+      }
+    };
 
     let serverOptions: ServerOptions = {
       run: {
-        command: cmd,
-        args: ["--disable-folding"]
+        module: this.buildPath(config, dir, false),
+        ...defaultModuleOptions,
+        args: runArgs
       },
       debug: {
-        command: cmd,
+        module: this.buildPath(config, dir, true),
```

Comment:
I think for debug we could just point to the file in node_modules, the copy should only be for release unless I am missing something there.

Reply:
Yeah, so in the debug mode. I would like use the current flux-lsp in the computer with pushing a branch. This will copy the flux-lsp wasm built to the dist directory. If I only use the node_modules dir, i have to make two branch, in both flux-lsp-cli and flux-lsp, and push the changes to both.

Reply:
No you don't I just clone the repo down for flux-lsp and flux-lsp-cli and then change vsflux's package.json to point to flux-lsp-cli on the file system and do the same for the flux-lsp in flux-lsp-cli.

Reply:
this is how I started building out new features for the lsp server and testing in vscode